### DISCUSSION TITLE
kafka CustomResourceDefinition

### DISCUSSION THREAD

**Discussion by zqw961025** [strimzi/strimzi-kafka-operator#7200] [Q&A]
I wonder if Kafka CRD will be added to the Kubernetes cluster after Kafka is installed with the modified project. If so, what exactly does CRD look like

**Comment by scholzj** [comment]
Can you please elaborate on what exactly you mean with this? To deploy Kafka with Strimzi you:
* need to have the CRD installed first
* The CRD is always the same, it does not change

**Reply by zqw961025** [reply]
I was just wondering how Kafka CRD is defined in Strimzi

**Reply by scholzj** [reply]
But what do you mean with _defined_? It is generated out of the Java classes which are in the `api` module if you mean how we create the YAML of it. We have our own CRD generator for it (see the `crd-generator` module in this repo).

**Reply by zqw961025** [reply]
I've found kafka crd here: https://github.com/strimzi/strimzi-kafka-operator/releases/download/0.30.0/strimzi-cluster-operator-0.30.0.yaml

**Reply by scholzj** [reply]
Yes, that is the generated YAML of the CRD.
